cvat/apps/engine/utils.py

```diff
@@ -83,6 +83,30 @@ def load_instances(model, primary_keys):
     return [instance_by_pk[pk] for pk in primary_keys]
 
 
+def find_range(iterable, predicate):
+    """Find the indices of the first range of consecutive items which satisfy the given predicate.
+    Returns (-1, -1) if it there is no such ranges.
+
+    find_range([0, 0, 1, 1, 0], lambda e: e > 0) => (2, 4)
+    """
+    start_index = -1
+    iterator = enumerate(iterable)
```

Comment:
consider
```
start_index = next(i for i, value in iterator if predicate(value))
end_index = next(i for i, value in iterator if not predicate(value))
```

Reply:
Works for `start_index`, doesn't work for `end_index` on this case: `find_range([0, 0, 1, 1], is_not_zero) == (2, 4)`  as I need to get the last value of `i` in this case.

Reply:
you get the gist, provide default as needed:
```
start_index = next((i for i, value in iterator if predicate(value)), -1)
end_index = next((i for i, value in iterator if not predicate(value)), len(iterable))
```

Reply:
Your version is fine, I am just surprised that you tend to use extremely complex Python syntax everywhere, but opted for basically C-style code for this trivial case.